eclipse-collections-code-generator/src/main/resources/test/list/immutable/abstractImmutablePrimitiveListTestCase.stg

```diff
@@ -288,6 +293,64 @@ public abstract class AbstractImmutable<name>ListTestCase extends AbstractImmuta
         Verify.assertEqualsAndHashCode(immutableList, list);
         Assert.assertNotSame(immutableList, list);
     }
+
+    @Test
+    public void zip<name>()
+    {
+        Immutable<name>List list1 = this.newWith(<["1", "2", "3"]:(literal.(type))(); separator=", ">);
+        Immutable<name>List list2 = this.newWith(<["1"]:(literal.(type))(); separator=", ">);
+        ImmutableList\<<name><name>Pair> zipSame = list1.zip<name>(list1);
+        Assert.assertEquals(
+                Lists.immutable.with(
+                    PrimitiveTuples.pair(<["1", "1"]:(literal.(type))(); separator=", ">),
+                    PrimitiveTuples.pair(<["2", "2"]:(literal.(type))(); separator=", ">),
+                    PrimitiveTuples.pair(<["3", "3"]:(literal.(type))(); separator=", ">)),
+                zipSame);
+        ImmutableList\<<name><name>Pair> zipLess = list1.zip<name>(list2);
+        Assert.assertEquals(
+                Lists.immutable.with(
+                    PrimitiveTuples.pair(<["1", "1"]:(literal.(type))(); separator=", ">)),
+                zipLess);
+        ImmutableList\<<name><name>Pair> zipMore = list2.zip<name>(list1);
+        Assert.assertEquals(
+                Lists.immutable.with(
+                    PrimitiveTuples.pair(<["1", "1"]:(literal.(type))(); separator=", ">)),
+                zipMore);
+        ImmutableList\<<name><name>Pair> zipEmpty1 = list1.zip<name>(this.newWith());
+        Assert.assertTrue(zipEmpty1.isEmpty());
+        ImmutableList\<<name><name>Pair> zipEmpty2 = this.newWith().zip<name>(list1);
+        Assert.assertTrue(zipEmpty2.isEmpty());
```

Comment:
There needs to be a test case for list1.size() = list2.size()

Reply:
Zip with the same list should cover this case.

Reply:
Ah! Sorry I missed that in this file.